utilities/imgcat

```diff
@@ -50,37 +52,68 @@ function b64_decode() {
     base64 $BASE64ARG
 }
 
-# print_image filename inline base64contents print_filename
+# print_image filename inline base64contents print_filename width height preserve_aspect_ratio
 #   filename: Filename to convey to client
-#   inline: 0 or 1
+#   inline: 0 or 1, if set to 1, the file will be displayed inline, otherwise, it will be downloaded
 #   base64contents: Base64-encoded contents
-#   print_filename: If non-empty, print the filename
-#                   before outputting the image
+#   print_filename: 0 or 1, if set to 1, print the filename after outputting the image
+#   width: set output width of the image in character cells, pixels or percent
+#   height: set output height of the image in character cells, pixels or percent
+#   preserve_aspect_ratio: 0 or 1, if set to 1, fill the specified width and height as much as possible without stretching the image
 function print_image() {
     print_osc
-    printf '1337;File='
-    if [[ -n $1 ]]; then
-        printf "name=%s;" "$(printf "%s" "$1" | b64_encode)"
-    fi
-
-    printf "%s" "$3" | b64_decode | wc -c | awk '{printf "size=%d",$1}'
-    printf ";inline=%s" "$2"
-    printf ":"
-    printf "%s" "$3"
+    printf "1337;File=inline=%s" "$2"
+    printf ";size=%d" $(printf "%s" "$3" | b64_decode | wc -c)
+    [ -n "$1" ] && printf ";name=%s" "$(printf "%s" "$1" | b64_encode)"
+    [ -n "$5" ] && printf ";width=%s" "$5"
+    [ -n "$6" ] && printf ";height=%s" "$6"
+    [ -n "$7" ] && printf ";preserveAspectRatio=%s" "$7"
+    printf ":%s" "$3"
     print_st
     printf '\n'
-    if [[ -n $4 ]]; then
-        echo "$1"
-    fi
+    [ "$4" == "1" ] && echo "$1"
+    has_image_displayed=t
 }
 
 function error() {
     echo "ERROR: $*" 1>&2
 }
 
+function errcho() {
+    echo "$@" >&2
+}
+
 function show_help() {
-    echo "Usage: imgcat [-p] filename ..." 1>&2
-    echo "   or: cat filename | imgcat" 1>&2
+    errcho
+    errcho "Usage: imgcat [-p] [-q] [-W width] [-H height] [-r] [-s] [-u] [-f] filename ..."
+    errcho "       cat filename | imgcat [-W width] [-H height] [-r] [-s]"
+    errcho
+    errcho "Display images inline in the iTerm2 using Inline Images Protocol"
+    errcho
+    errcho "Options:"
+    errcho
+    errcho "    -h, --help                      Display help message"
+    errcho "    -p, --print                     Enable printing of filename or URL after displaying each image"
+    errcho "    -q, --quiet                     Be quiet, disable filename printing (default)"
```

Comment:
If this is the default how do you disable it? Maybe it should be `-v, --verbose` instead.

Reply:
Maybe it is not properly named. Option 'quiet' is not about to be totally quiet concerning error messages and something like this. It is only about filename printing.

 - `-p` - enable filename printing at the end of each image
 - `-q` - disable filename printing at the end of each image

Also see my explanation about command-line arg processing below.